fastai/gen_doc/doctest.py

```diff
@@ -35,33 +35,34 @@ def this_tests(*funcs):
             else:
                 raise Exception(f"'{func}' is not in the fastai API")
         try: 
-            TestAPIRegistry.failed_tests.remove(TestAPIRegistry.has_this_tests) 
+            missing_this_test = 'file: '+ relative_test_path(filename) + ' / test: ' + test_name
+            TestAPIRegistry.missing_this_tests.remove(missing_this_test) 
         except:
             None
         TestAPIRegistry.has_this_tests = None
 
-    def this_tests_flag_on(item):
-        TestAPIRegistry.has_this_tests = item.name
-        ## to do: get fully qualifed name from item.name & then find filename. place in dictionary
-        TestAPIRegistry.failed_tests.add(TestAPIRegistry.has_this_tests)
+    def this_tests_flag_on(filename, test_name):
+        TestAPIRegistry.has_this_tests = test_name
+        missing_this_test = 'file: '+ filename + ' / test: ' + test_name
+        TestAPIRegistry.missing_this_tests.add(missing_this_test)
        
     def tests_failed(status=True):
         TestAPIRegistry.some_tests_failed = status
     
-    def this_tests_flag_check(item):
-        if TestAPIRegistry.has_this_tests == item.name:
+    def this_tests_flag_check(filename, test_name):
+        if TestAPIRegistry.has_this_tests == test_name:
             TestAPIRegistry.has_this_tests = None
 
     def registry_save():
+        if TestAPIRegistry.missing_this_tests:
+            print('\n*** Warning: Pls register the following tests with TestAPIRegistry.this_tests:\n')
```

Comment:
too complicated - please use join() here - all of it should be one command.

Reply:
dont understand

Reply:
I suppose you mean sth like this: print('\n'.join(url.format(year) for year in range(2000, 2016))) ?

Reply:
print('\n'.join(str(missing_this_test) for missing_this_test in TestAPIRegistry.missing_this_tests))